contrib/experimental/great_expectations_experimental/expectations/expect_column_values_to_be_valid_currency_code.py

```diff
@@ -0,0 +1,142 @@
+import json
+from typing import Optional, Tuple
+
+from moneyed import list_all_currencies
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.exceptions import InvalidExpectationConfigurationError
+from great_expectations.execution_engine import (
+    PandasExecutionEngine,
+    SparkDFExecutionEngine,
+    SqlAlchemyExecutionEngine,
+)
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+def generate_all_currency_codes() -> Tuple[str]:
+    return [str(currency_code) for currency_code in list_all_currencies()]
+
+
+def is_valid_currency_code(code: str, currency_codes: Tuple[str]) -> bool:
+    return code in currency_codes
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesCurrencyCode(ColumnMapMetricProvider):
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.currency_code"
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column, **kwargs):
+
+        currency_codes: Tuple[str] = generate_all_currency_codes()
+
+        return column.apply(lambda x: is_valid_currency_code(x, currency_codes))
```

Comment:
Wouldn't be a good idea to uppercase `x` before comparing? so that the case of `usd` and `USD` are covered?

Reply:
I thought about that as well. I'm thinking the best option would be to make it an optional param like "strict_case". That way if someone actually wanted the column values to be formal entries like `USD` instead of `usd`, it'd cover that.

Reply:
I believe `usd` doesn't qualify as a valid currency code, per the ISO-4217 this expectation is targeting.

Reply:
Yep. I suppose if we hear feedback from any users that a non-strict option would be useful, it could be discussed.